domain/data/tuning.py

```diff
@@ -0,0 +1,161 @@
+import os
+import sys
+
+root_dir = 'system/root/directory'
+sys.path.append(root_dir)
+
+import argparse
+from typing import List
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import lightning.pytorch as pl
+import torch.optim as optim
+import optuna
+from optuna.integration import PyTorchLightningPruningCallback
+from torchvision import datasets, transforms
+from domain.model.modules import MLP, ResidualMLP, MLP_Block_Residual, ResCNN, CNN, ResCNN_block, VisionTransformer
+from domain.data.data_modules import MNISTDataModule, CIFAR10DataModule, ImageNetDataModule
+from domain.model.lit_modules import LitClassificationModel as LiT
+
+PERCENT_VALID_EXAMPLES = 0.1
+BATCHSIZE = 128
```

Comment:
The default value for batch size may be too big. Also, we should take these as arguments. Later, we may want to tune it.